docs/notebooks/bald.pct.py

```diff
@@ -0,0 +1,144 @@
+# %%
+from dataclasses import astuple
+
+import gpflow
+import matplotlib.pyplot as plt
+import numpy as np
+import tensorflow as tf
+from gpflow.utilities import print_summary, set_trainable
+
+import trieste
+from trieste.acquisition.function import BALD, PredictiveVariance
+from trieste.acquisition.rule import OBJECTIVE
+from trieste.models import create_model
+from trieste.models.gpflow.models import GaussianProcessRegression, VariationalGaussianProcess
+from trieste.objectives.utils import mk_observer
+from trieste.utils import map_values
+
+np.random.seed(1965)
+tf.random.set_seed(1965)
+
+sp = trieste.space.Box([-1, -1], [1, 1])
+inDim = 2
+
+
+def circle(x):
```

Comment:
is this a common toy problem in this literature?
what about replicating the problems from the paper (results from Figure 3 - only the BALD and perhaps random search)?

Reply:
not sure if this function is common or not,
The reason we use this toy problem is that, it is more illustrative than for example calculating the accuracy like in the paper. I.e we can see how BALD can sample near the boundary region of the two classes in the circle dataset. In addition, this makes us easily use the continuous search space and EGO acquisition rule. 
What do you think?

Reply:
that's a fair point, but it would still be nice to use a toy problem that has been used in the literature, as widely as possible - I cannot provide much guidance here unfortunately, perhaps @vpicheny ?